lang/funcs/engine.go

```diff
@@ -45,7 +45,7 @@ type State struct {
 	input  chan types.Value // the top level type must be a struct
 	output chan types.Value
 
-	mutex *sync.RWMutex // concurrency guard for modifying Expr with String/SetValue
+	mutex *sync.RWMutex // concurrency guard for accessing Expr.String()
 }
 
 // Init creates the function state if it can be found in the registered list.
```

Comment:
github won't let me attach a comment to a line which hasn't changed, but a few dozen lines below, at line 82, there's another occurrence of `SetValue` in a comment, and I'm not sure how to handle it:

```go
// String satisfies fmt.Stringer so that these print nicely.
func (obj *State) String() string {
	// TODO: use global mutex since it's harder to add state specific mutex
	//obj.mutex.RLock() // prevent race detector issues against SetValue
	//defer obj.mutex.RUnlock()
	// FIXME: also add read locks on any of the children Expr in obj.Expr
	return obj.Expr.String()
}
```

It's a commented-out line anyway, so I should probably just delete it, but the TODO next to it seems related so I'm not sure whether I should delete it too or if I would accidentally be removing an important TODO note.